src/api/auth.ts

```diff
@@ -0,0 +1,91 @@
+import {callAPI} from "./callApi";
+
+const generateState = (prefix: string) => {
+  // generate random state id and store origin into the session storage
+  const state = `${prefix}-${Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15)}`;
```

Comment:
You should split these parts for better readability. Also explain why you chose these numbers in a comment ;)

Reply:
What would you like to split? The random string blocks? As for explanation: in toString(36): 36 is the "radix – An integer in the range 2 through 36 specifying the base to use for representing numeric values." --> giving you a mix of digit and word characters. In substring(2, 15): Beginning with 2, otherwise you have 0. in the beginning; 15 just because it seems to be a common way to create random strings (e.g. https://gist.github.com/6174/6062387)

Reply:
That is correct. But it is still good to add this as a comment in the code. But that is up to you. If you think this is not required (without looking for common ways to create a uuid), that's fine for me too :)

Reply:
you're absolutely right about a common way to create uuids. I swapped the custom method above for uuidv4().